### PR TITLE
Feature/web app representations get by csv

### PR DESCRIPTION
**Pull request by jonathangoulding**
## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: apps/web/src/server/applications/case/representations/download/download.controller.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import got from 'got';
+
+export const getRepDownloadController = async (req, res) =>
+	got
+		.stream(`http://localhost:3000/applications/${req.params.caseId}/representations/download`)
```
Comment (jonathangoulding) [inline apps/web/src/server/applications/case/representations/download/download.controller.js:5]: This needs to use the utils got if possible.

File: apps/api/src/server/applications/application/representations/download/rep-download.controller.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+export const getCaseRepDownload = async (req, res) => {
+	const readableStream = new Readable({ objectMode: true });
+	const transformToCSV = new TransformToCSV({ objectMode: true });
+
+	const batchSize = 100;
+	getCaseRepresentationToDownloadInBatches(readableStream, batchSize, Number(req.params.id));
+
+	readableStream._read = () => {};
+
+	res.set('Content-disposition', 'attachment; filename=' + 'output.csv');
```
Comment (jonathangoulding) [inline apps/api/src/server/applications/application/representations/download/rep-download.controller.js:14]: file name needs to be changed.

Comment (jonathangoulding) [reply apps/api/src/server/applications/application/representations/download/rep-download.controller.js:14]: relevant-reps-<case ref no>-<date now>

**Review by Philip-Philippou1 — APPROVED** [review]

File: apps/api/src/server/applications/application/representations/download/rep-download.controller.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+	const caseId = Number(req.params.id);
+
+	const batchSize = 100;
+	getCaseRepresentationToDownloadInBatches(readableStream, batchSize, caseId);
+
+	readableStream._read = () => {};
+
+	res.set(
+		'Content-disposition',
+		'attachment; filename=' + `relevant-reps-${caseId}-${Date.now()}.csv`
```
[the remark this answers is not in the document]
Comment (jonathangoulding) [reply apps/api/src/server/applications/application/representations/download/rep-download.controller.js:17]: updated to produce - 
relevant-reps-1301-2023-07-25-15-50.csv

**Review by Philip-Philippou1 — APPROVED** [review]

**Pull request merged**